### PR TITLE
Refactor with hooks

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
In this PR I:
- Refactored `Calculator.js` and `Button.js` from a class based component to a functional component with hooks.

### CODE REVIEW

File: src/app/components/Button.js [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 import { v4 as uuidv4 } from 'uuid';
+import PropTypes from 'prop-types';
 
-export default class Button extends react.Component {
-  constructor(props) {
-    super(props);
-    this.btn = this.props.btn;
-  }
+const Button = ({ click }) => {
+  const btn = ['AC', '+/-', '%', '+', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', ' ='];
```
Comment (TedLivist) [inline src/app/components/Button.js]: - [x] Your app has 2 plus (+) signs and as a result, is missing the division sign (&#247;). Kindly fix this.

  **SUGGESTION:** Here is the HTML entity code for the sign if you wish to use it: **`\u00F7`**. To resolve this you can change this line to have something like so:
  `const btn = ['AC', '+/-', '%', '\u00F7', 7, 8, 9, 'x', 4, 5, 6, '-', 1, 2, 3, '+', 0, '.', ' ='];`

  sahar-plus

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Sahar-AbdelSamad,

Good job so far!
✔️ Calculator component is refactored to function component.
✔️ Use of hooks
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] Let's try to be thorough in our documentation as it can be of great benefit to users. Your link for cloning the repo, as contained in the README is incorrect. You used hyphen ( - ), instead of the underscore ( _ ). Here is the correct link: **`git@github.com:Sahar-AbdelSamad/math_magicians.git`**

  sahar-README

- [x] **_Check the comments under the review._**

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app/components/Button.js (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply src/app/components/Button.js]: Thank you so much for your review and your valuable comments :smile:  I fixed these issues :+1:

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Sahar-AbdelSamad,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**